docs/tui.md

```diff
@@ -0,0 +1,234 @@
+  - Prometheus (MC) and Grafana (MC) always connect to the Management Cluster
+  - Alloy Metrics connects to the Workload Cluster if one is specified, otherwise it connects to the Management Cluster
+- `portForwardProcess` struct: Tracks process state (including which cluster it targets), output, and errors for each port-forward.
+- Status update messages: Keep the UI in sync with actual process status for all services.
+
+### Context Switching
+
+The TUI handles context switching through:
+- `connection_flow.go`: Functions to manage the connection flow
+- `handlers.go`: Event handlers for keyboard shortcuts
+- Asynchronous operations to update the UI as contexts change
+
+### Viewport Management
+
+Scrollable log views are implemented using Bubble Tea's viewport component:
+- `mainLogViewport`: For the in-line log panel
+- `logViewport`: For the expandable log overlay
+- Mouse wheel scrolling support in both viewports
+
+### Debug Features
+
+The TUI includes debugging capabilities:
+- Toggle debug mode with 'z' key
+- View raw dimensions and layout calculations
+- Detect and display color profile information
+
+## File Structure
+
+- `model.go`: Core model definition, Update() and View() methods
+- `view_helpers.go`: Helper functions for rendering UI components
+- `handlers.go`: Key press and event handlers
+- `styles.go`: UI styling definitions
+- `portforward_handlers.go`: Port forwarding logic
+- `connection_flow.go`: Cluster connection handling
+- `message_types.go`: Custom message type definitions
+
+## Design Decisions
+
+### Use of Bubble Tea & Lipgloss
+
+These libraries were chosen for:
+- Strong typing and Go-native approach
+- Excellent terminal compatibility
+- Component composition model
+- Elegant handling of async events
+
+### Separate UI and Logic Components
+
+The codebase separates:
+- UI rendering (view helpers)
+- State management (model)
+- Business logic (handlers)
+
+This separation enables easier testing and maintenance.
+
+### Asynchronous Communication
+
+The TUI uses channels for non-blocking operations to:
+- Keep the UI responsive while long-running operations execute
+- Allow real-time updates of port forward status
+- Support health checking in the background
+
+## Troubleshooting
+
+### Common Issues
+
+- **Layout Issues**: If panels appear misaligned, it may be due to terminal font settings or Unicode rendering
+- **Color Problems**: Some terminals may not support all colors; use 'D' to toggle between modes
+- **Performance**: Large log output can impact performance; consider increasing buffer size if needed
+
+### Debugging
+
+1. Enable debug mode with 'z' key
+2. Check terminal dimensions and layout calculations
+3. Review logs for any error messages
+
+## Future Enhancements
+
+- Clickable UI elements for easier navigation
+- Additional panel types for other service statuses
+- Draggable/resizable panels
+- Configuration options for colors and layout
+- Search functionality in logs 
```

Comment:
The documentation mentions mouse wheel scrolling support, but this doesn't appear to be implemented in the code. Either implement the feature or remove it from the documentation to avoid confusion.